phoenix-core/src/main/java/org/apache/phoenix/calcite/CalciteUtils.java

```diff
@@ -1036,4 +1048,40 @@ public Void visitCall(RexCall call) {
             return null;
         }
     }
+
+    public static Object convertLiteral(SqlLiteral literal, PhoenixRelImplementor implementor) {
+        try {
+            final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+            final FrameworkConfig config = Frameworks.newConfigBuilder()
+                    .parserConfig(SqlParser.Config.DEFAULT)
+                    .defaultSchema(rootSchema).build();
+            Planner planner = Frameworks.getPlanner(config);
+
+            SqlParserPos POS = SqlParserPos.ZERO;
+            final SqlNodeList selectList =
+                    new SqlNodeList(
+                            Collections.singletonList(literal),
+                            SqlParserPos.ZERO);
+
+
+            String sql = new SqlSelect(POS, SqlNodeList.EMPTY, selectList, null, null, null, null,
```

Comment:
Thank you, Eric, for doing a nice job and trying to work things out in the best you could find! But if you run into issues like this next time, you can just ping me quickly. I can probably give you some useful information since I've had quite some experience dealing with Calcite.